backend/tests/unit/dal/Services/ProjectServiceTest.cs

```diff
@@ -367,6 +367,27 @@ public void Get_Sensitive_AsAdmin()
         }
         #endregion
 
+        # region Generates ProjectNumber
+        [Fact]
+        public void Generates_ProjectNumber()
+        {
+            //Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.PropertyAdd).AddAgency(1);
+
+            var dbName = StringHelper.Generate(10);
+            using var init = helper.InitializeDatabase(dbName, user);
+            var project = init.CreateProject(1);
+            init.SaveChanges();
+
+            // Act 
+            
```

Comment:
Your Act is missing.  You should be calling the `service.Add(...)`
You are currently testing our test code, not the DAL with this.